src/Form/ConnectionSettingsForm.php

```diff
@@ -145,6 +146,15 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     return parent::buildForm($form, $form_state);
   }
 
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    parent::validateForm($form, $form_state);
+
+    // Check if APM server URL is valid.
+    if (!UrlHelper::isValid($form_state->getValue('server_url'), TRUE)) {
+      $form_state->setErrorByName('server_url', $this->t('The server URL is invalid.'));
+    }
+  }
```

Comment:
**CHANGE REQUEST**

What this is issue is about is to actually check that the connection works and the credentials are valid. Check if the SDK provides such a function.